Shared/Extensions/WKWebViewExtensions.swift

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import JavaScriptCore
+import WebKit
+
+enum JavascriptError: Error {
+    case invalid
+}
+
+extension WKUserContentController {
+    public func addInDefaultContentWorld(scriptMessageHandler: WKScriptMessageHandler, name: String) {
+        if #available(iOS 14.0, *) {
+            add(scriptMessageHandler, contentWorld: .defaultClient, name: name)
+        } else {
+            add(scriptMessageHandler, name: name)
+        }
+    }
+}
+
+extension WKUserScript {
+    public class func createInDefaultContentWorld(source: String, injectionTime: WKUserScriptInjectionTime, forMainFrameOnly: Bool) -> WKUserScript {
+        if #available(iOS 14.0, *) {
+            return WKUserScript(source: source, injectionTime: injectionTime, forMainFrameOnly: forMainFrameOnly, in: .defaultClient)
+        } else {
+            return WKUserScript(source: source, injectionTime: injectionTime, forMainFrameOnly: forMainFrameOnly)
+        }
+    }
+}
+
+public extension WKWebView {
+    func generateJavascriptFunctionString(functionName: String, args: [Any], escapeArgs: Bool = true) -> (String, Error?) {
+        let context = JSContext()
+
+        var sanitizedArgs: [String] = []
+        var error: Error?
+
+        args.forEach() {
+            if !escapeArgs {
+                return sanitizedArgs.append("\($0)")
+            }
+            context?.exceptionHandler = { context, exception in
+                if exception != nil {
+                    error = JavascriptError.invalid
+                }
+            }
+            context?.evaluateScript("JSON.parse('\"\($0)\"')")
+            return sanitizedArgs.append("'\(String(describing: $0).escapeHTML())'")
+        }
+        
+        return ("\(functionName)(\(sanitizedArgs.joined(separator: ", ")))", error)
+    }
+
+    func evaluateSafeJavascript(functionName: String, args: [Any], sandboxed: Bool = true, escapeArgs: Bool = true, asFunction: Bool = true, completion: ((Any?, Error?) -> Void)?) {
+        var javascript = functionName
+        
+        if asFunction {
+            let ret = generateJavascriptFunctionString(functionName: functionName, args: args, escapeArgs: escapeArgs)
+            if ret.1 != nil {
```

Comment:
Use named parameters, `0` and `1` are unreadable

Reply:
Updated to use named parameters